developer-extension/src/panel/hooks/useEvents.ts

```diff
@@ -22,17 +22,26 @@ export function useEvents() {
     query: '',
   })
 
-  useEffect(
-    () =>
-      listenRequests((newEvents) => {
-        setEvents((oldEvents) =>
-          [...newEvents, ...oldEvents]
-            .sort((first: any, second: any) => second.date - first.date)
-            .slice(0, MAXIMUM_LOGGED_EVENTS)
-        )
-      }),
-    []
-  )
+  useEffect(() => {
+    const removeListener = listenRequests((newEvents) => {
+      setEvents((oldEvents) =>
+        [...newEvents, ...oldEvents]
+          .sort((first: any, second: any) => second.date - first.date)
+          .slice(0, MAXIMUM_LOGGED_EVENTS)
+      )
+    })
+
+    const clearCurrentEvents = (details: chrome.webNavigation.WebNavigationTransitionCallbackDetails) => {
+      if (['reload'].includes(details.transitionType)) setEvents([])
+    }
+
+    chrome.webNavigation.onCommitted.addListener(clearCurrentEvents)
+
+    return () => {
+      removeListener()
+      chrome.webNavigation.onCommitted.removeListener(clearCurrentEvents)
+    }
+  }, [])
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Sometimes it could be usefull to "preserve" events accross refresh. You could do something like the "auto flush" PR https://github.com/DataDog/browser-sdk/pull/1824 by showing a checkbox to "Preserve events". The `useEvents` hook could look like this:


```ts
function useEvents(preserveEvents: boolean) {
  // ...

  useEffect(() => {
    if (!preserveEvents) {
       const clearCurrentEvents = (details: chrome.webNavigation.WebNavigationTransitionCallbackDetails) => {
         if (details.transitionType === 'reload') setEvents([])
      }
      chrome.webNavigation.onCommitted.addListener(clearCurrentEvents)
      return () => {
        chrome.webNavigation.onCommitted.removeListener(clearCurrentEvents)
      }
    }
  }, [preserveEvents])
  
  // ...
}
```